psuutil/main.py

```diff
@@ -0,0 +1,192 @@
+
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+
+# ==================== Methods for initialization ====================
+
+# Returns platform and HW SKU
+def get_platform_and_hwsku():
+    try:
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-v', PLATFORM_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        platform = stdout.rstrip('\n')
+
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-m', MINIGRAPH_PATH, '-v', HWSKU_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        hwsku = stdout.rstrip('\n')
+    except OSError, e:
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+
+# Loads platform specific psuutil module from source
+def load_platform_psuutil():
+    global platform_psuutil
+
+    # Get platform and hwsku
+    (platform, hwsku) = get_platform_and_hwsku()
+
+    # Load platform module from source
+    platform_path = ''
+    if len(platform) != 0:
+        platform_path = "/".join([PLATFORM_ROOT_PATH, platform])
+    else:
+        platform_path = PLATFORM_ROOT_PATH_DOCKER
+    hwsku_path = "/".join([platform_path, hwsku])
+
+    try:
+        module_file = "/".join([platform_path, "plugins", PLATFORM_SPECIFIC_MODULE_NAME + ".py"])
+        module = imp.load_source(PLATFORM_SPECIFIC_MODULE_NAME, module_file)
+    except IOError, e:
+        log_error("Failed to load platform module '%s': %s" % (PLATFORM_SPECIFIC_MODULE_NAME, str(e)), True)
+        return -1
+
+    try:
+        platform_psuutil_class = getattr(module, PLATFORM_SPECIFIC_CLASS_NAME)
+        platform_psuutil = platform_psuutil_class()
+    except AttributeError, e:
+        log_error("Failed to instantiate '%s' class: %s" % (PLATFORM_SPECIFIC_CLASS_NAME, str(e)), True)
+        return -2
+
+    return 0
+
+
+# ==================== CLI commands and groups ====================
+
+
+# This is our main entrypoint - the main 'psuutil' command
+@click.group()
+def cli():
+    """psuutil - Command line utility for providing PSU status"""
+
+    if os.geteuid() != 0:
+        print "Root privileges are required for this operation"
+        sys.exit(1)
+
+    # Load platform-specific psuutil class
+    err = load_platform_psuutil()
+    if err != 0:
+        sys.exit(2)
+
+# 'version' subcommand
+@cli.command()
+def version():
+    """Display version info"""
+    click.echo("psuutil version {0}".format(VERSION))
+
+# 'numpsus' subcommand
+@cli.command()
+def numpsus():
+    "number of supported PSU in the device"
```

Comment:
Suggest `"Display number of supported PSU in the device"`

Reply:
Done